astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -23,212 +168,638 @@ class NDArithmeticMixin(object):
     When subclassing, be sure to list the superclasses in the correct order
     so that the subclass sees NDData as the main superclass. See
     `~astropy.nddata.NDDataArray` for an example.
+
+    Notes
+    -----
+    This class only aims at covering the most common cases so there are certain
+    restrictions on the saved attributes::
+
+        - ``uncertainty`` : has to be something that has a `NDUncertainty`-like
```

Comment:
Correct me if I'm wrong - I believe that the idea was that it is an arbitrary object that has an attribute `.uncertainty_type`